staging/operator-lifecycle-manager/pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -978,58 +997,93 @@ func (a *Operator) updateNamespaceList(op *operatorsv1.OperatorGroup) ([]string,
+	nameLimit := kubeResourceNameLimit - len(strings.Replace(operatorGroupClusterRoleNameFmt, "%s", "", -1)) - len(roleType) - len(roleSuffix)
+	if len(op.GetName()) < nameLimit {
+		return fmt.Sprintf(operatorGroupClusterRoleNameFmt, op.GetName(), roleType, roleSuffix), nil
 	}
-	if len(selectors) > 0 {
-		clusterRole.AggregationRule = &rbacv1.AggregationRule{
-			ClusterRoleSelectors: selectors,
-		}
+	return fmt.Sprintf(operatorGroupClusterRoleNameFmt, op.GetName()[:nameLimit], roleType, roleSuffix), nil
+}
+
+func (a *Operator) ensureOpGroupClusterRole(op *operatorsv1.OperatorGroup, suffix string, apis cache.APISet) error {
+	// create target cluster role spec
+	var clusterRole *rbacv1.ClusterRole
+	clusterRoleName, err := a.getClusterRoleName(op, suffix)
+	if err != nil {
+		return err
 	}
-	err := ownerutil.AddOwnerLabels(clusterRole, op)
+	aggregationRule, err := a.getClusterRoleAggregationRule(apis, suffix)
 	if err != nil {
 		return err
 	}
 
-	existingRole, err := a.lister.RbacV1().ClusterRoleLister().Get(clusterRole.Name)
+	// get existing cluster role for this level (suffix: admin, edit, view))
+	existingRole, err := a.lister.RbacV1().ClusterRoleLister().Get(clusterRoleName)
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	if apierrors.IsNotFound(err) {
-		existingRole, err = a.opClient.KubernetesInterface().RbacV1().ClusterRoles().Create(context.TODO(), clusterRole, metav1.CreateOptions{})
-		if err == nil {
+
+	if existingRole != nil {
+		// if the existing role conforms to the naming convention, check for skew
+		cp := existingRole.DeepCopy()
+		if err := ownerutil.AddOwnerLabels(cp, op); err != nil {
+			return err
+		}
+
+		// ensure that the labels and aggregation rules are correct
+		if labels.Equals(existingRole.Labels, cp.Labels) && equality.Semantic.DeepEqual(existingRole.AggregationRule, aggregationRule) {
 			return nil
 		}
-		if !apierrors.IsAlreadyExists(err) {
-			a.logger.WithError(err).Errorf("Create cluster role failed: %v", clusterRole)
-			return err
+
+		cp.AggregationRule = aggregationRule
+		if _, err := a.opClient.UpdateClusterRole(cp); err != nil {
+			a.logger.WithError(err).Errorf("update existing cluster role failed: %v", clusterRole)
 		}
+		return err
 	}
 
-	if existingRole != nil && labels.Equals(existingRole.Labels, clusterRole.Labels) && reflect.DeepEqual(existingRole.AggregationRule, clusterRole.AggregationRule) {
-		return nil
+	clusterRole = &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterRoleName,
+		},
+		AggregationRule: aggregationRule,
 	}
 
-	if _, err := a.opClient.UpdateClusterRole(clusterRole); err != nil {
-		a.logger.WithError(err).Errorf("Update existing cluster role failed: %v", clusterRole)
+	if err := ownerutil.AddOwnerLabels(clusterRole, op); err != nil {
 		return err
 	}
-	return nil
+
+	a.logger.Infof("creating cluster role: %s owned by operator group: %s/%s", clusterRole.GetName(), op.GetNamespace(), op.GetName())
```

Comment:
Is this log intentional?

Reply:
Yes. Generally, more logs are better? This is particularly useful as there are two naming schemes.

Reply:
As long as we aren't blowing this up the logs this is fine.

Reply:
It only applies on _creation_